### PR TITLE
CoreFX: Add System.Transactions

### PR DESCRIPTION
**Pull request by mattleibow**
This is a PR for: https://github.com/mono/mono/issues/6972

### CODE REVIEW

File: mcs/class/System.Transactions/System.Transactions.csproj [hunk 1]
```diff
@@ -112,44 +112,11 @@
...
-    <Compile Include="System.Transactions\TransactionManagerCommunicationException.cs" />
-    <Compile Include="System.Transactions\TransactionOptions.cs" />
-    <Compile Include="System.Transactions\TransactionPromotionException.cs" />
-    <Compile Include="System.Transactions\TransactionScope.cs" />
-    <Compile Include="System.Transactions\TransactionScopeAsyncFlowOption.cs" />
-    <Compile Include="System.Transactions\TransactionScopeOption.cs" />
-    <Compile Include="System.Transactions\TransactionStatus.cs" />
+    <Compile Include="corefx\SR.cs" />
+    <Compile Include="..\..\..\external\corefx\src\System.Transactions.Local\src\System\Transactions\*.cs" />
+    <Compile Include="..\..\..\external\corefx\src\System.Transactions.Local\src\System\Transactions\Configuration\*.cs" />
```
[the remark this answers is not in the document]
Comment (EgorBo) [reply mcs/class/System.Transactions/System.Transactions.csproj]: @mattleibow we usually don't touch *csproj and when a PR is ready we do `@monojenkins commit csproj` (example - https://github.com/mono/mono/pull/11944#issuecomment-449326399) to update all the csprojs

**Comment by EgorBo** [comment]
There are some missing APIs (System.Transactions.Configuration.*) and probably some potential binary serialization issues (missing `ISerializable`)

**Comment by mattleibow** [comment]
@EgorBo I had a look and it appears that the serialisation stuff was moved into the base `Transaction` type - and also throws. With regards to the configuration, I am not sure what the process here is - it is marked internal, but should be public.

**Comment by marek-safar** [comment]
This assembly is quite Mono specific and fixing it at this point is not worth it. If you plan re-implementing the missing pieces please reopen the issue

**Pull request closed without merging**